lib/CXGN/Phenotypes/StorePhenotypes.pm

```diff
@@ -537,147 +550,164 @@ sub store {
                 my $trait_cvterm = $trait_objs{$trait_name};
 
                 my $value_array = $plot_trait_value{$plot_name}->{$trait_name};
-                my $trait_value = $value_array->[0];
-                my $timestamp = $value_array->[1];
-                $operator = $value_array->[2] ? $value_array->[2] : $operator;
-                my $observation = $value_array->[3];
-                my $image_id = $value_array->[4];
-                my $additional_info = $value_array->[5] || undef;
-                my $external_references = $value_array->[6] || undef;
-                my $unique_time = $timestamp && defined($timestamp) ? $timestamp : 'NA'.$upload_date;
-
-                if (defined($trait_value) && length($trait_value)) {
-
-                    if ($ignore_new_values) {
-                        if (exists($check_unique_trait_stock{$trait_cvterm->cvterm_id(), $stock_id})) {
-                            $skip_count++;
-                            next;
+
+                my @values;
+
+                # convert to array or array format for single array values to accept old format inputs without refactoring
+                if (ref($value_array->[0]) ne 'ARRAY') {
+                    push @values, $value_array;
+                } else {
+                    @values = @{$value_array};
+                }
+
+                foreach my $value (@values) {
```

Comment:
Breedbase was only saving one observation per observation unit/variable combo regardless of different timestamps within a single POST /observations request.